erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -36,6 +40,14 @@ def before_submit(self):
 				continue
 			frappe.throw(_('For item {0} at row {1}, count of serial numbers does not match with the picked quantity')
 				.format(frappe.bold(item.item_code), frappe.bold(item.idx)))
+	
+	def validate_delivery_date(self):
+		so = frappe.get_doc("Sales Order", self.locations[0].get("sales_order")).as_dict()
```

Comment:
Why only the first row? Also, check if there's a value in the first place, and maybe an additional check for the Pick List type to be "Delivery against Sales Order".